covalent/executor/base.py

```diff
@@ -67,10 +67,10 @@ class BaseExecutor(ABC):
 
     def __init__(
         self,
-        log_stdout: Optional[str] = "",
-        log_stderr: Optional[str] = "",
-        conda_env: Optional[str] = "",
-        cache_dir: Optional[str] = "",
+        log_stdout: str = "",
+        log_stderr: str = "",
+        conda_env: str = "",
+        cache_dir: str = "",
```

Comment:
Why get rid of the `Optional` annotation?

Reply:
@sayandipdutta I know I agreed with it in the comment https://github.com/AgnostiqHQ/covalent/issues/62#issuecomment-1027493779, my bad, I realize now that removing `Optional` here is going to be non-standard. Since, the purpose of `Optional` is to indicate that this parameter already has a default value irrespective of whether it is an empty string or not.

Reply:
Great, I'll approve once this is put back in and we are back up-to-date with develop.

Reply:
I can put back `Optional`, just would request you to check what actual `typing.Optional` docs say about default value type annotations. https://docs.python.org/3/library/typing.html#typing.Optional
> Note that this is not the same concept as an optional argument, which is one that has a default. An optional argument with a default does not require the Optional qualifier on its type annotation just because it is optional.

Reply:
And this is not merely due to aesthetics, in my experience, `Optional` qualifiers can create some annoying type checking issues with mypy. For example:
**test.py**
```python
from typing import Optional
def foo(arg: str) -> None:
    pass

def bar(arg: Optional[str] = "") -> None:
    foo(arg)
    return
```
Running `mypy`:
```python
> mypy .\test.py
.\test.py:6: error: Argument 1 to "foo" has incompatible type "Optional[str]"; expected "str"
Found 1 error in 1 file (checked 1 source file)
```
Then you will need to do one of the following:
```python
from typing import cast
def bar(arg: Optional[str] = "") -> None:
    foo(cast(str, arg))
    return
```
**or**
```python
def bar(arg: Optional[str] = "") -> None:
    assert arg is not None
    foo(arg)
    return
```
**or**
```python
def bar(arg: Optional[str] = "") -> None:
    if isinstance(arg, str):
        foo(arg)
    return
```
Although a single `cast` , `assert` or `isinstance` may not be a big deal, if you have to use it repeatedly, especially in loops, it looks unclean and reduces performance.

Reply:
Ok, sounds good. Can you pull in the latest changes from develop, and update the VERSION and CHANGELOG files?